src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -4825,6 +4826,19 @@ def _apply_guardrail_with_retry(self, response_text, prompt, temperature=1.0, to
             retry_count += 1
             logging.warning(f"Agent {self.name}: Guardrail validation failed (retry {retry_count}/{self.max_guardrail_retries}): {error}")
             
+            # Add exponential backoff delay to avoid hammering the LLM
+            execution_config = getattr(self, 'execution_config', None)
+            if execution_config is not None:
+                delay = execution_config.retry_initial_delay * (execution_config.retry_backoff_factor ** (retry_count - 1))
+                jitter = random.uniform(0, execution_config.retry_jitter * delay)
```

Ruff (0.15.15)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)